feat: Support for arbitrary versions of z3 (#2820)
dafny-lang/dafny: docs/DafnyRef/Options.txt

  
  /extractCounterexample
      If verification fails, report a detailed counterexample for the
      first failing assertion. Requires specifying the /mv:<file> option as well

>       as /proverOpt:O:model_compress=false (for z3 version < 4.8.7)" or /proverOpt:O:model.compact=false (for z3 version >= 4.8.7), and /proverOpt:O:model.completion=true.

Author: ```suggestion
      as /proverOpt:O:model_compress=false (for z3 version < 4.8.7) or /proverOpt:O:model.compact=false (for z3 version >= 4.8.7), and /proverOpt:O:model.completion=true.
```

  
  /extractCounterexample
      If verification fails, report a detailed counterexample for the

>       first failing assertion. Requires specifying the /mv:<file> option as well
>       as /proverOpt:O:model_compress=false (or /proverOpt:O:model.compact=false if
>       your version of z3 is >= 4.8.7) and /proverOpt:O:model.completion=true.

Author: ```suggestion
      first failing assertion. Requires specifying the /mv:<file> option as well
      as /proverOpt:O:model_compress=false (for z3 version < 4.8.7)" or /proverOpt:O:model.compact=false (for z3 version >= 4.8.7), and /proverOpt:O:model.completion=true.
```
